### DISCUSSION TITLE
Vers. 0.4.2 - major update for macOS

### DISCUSSION THREAD

**Discussion by kaixxx** [kaixxx/noScribe#32] [Announcements]
This update solves two major problems in macOS:
- fixed: speaker-detection was unreliable with Apple MPS-acceleration (a switch to torchaudio 2.1.0 rectified this).
- fixed: noScribe under macOS Sonoma was often not reacting to the mouse (caused by a conflict between Tcl/Tk an Sonoma). 

The windows-version is unchanged.

**Comment by gernophil** [comment]
Hey everyone,

seems like with using torchaudio 2.1.0 we introduced some new bugs.

I did some tests on a different MacBook then mine (normall I only have one for testing). The arm64 version suffers from frequent out-of memory errors (see screenshot 1).

The x86_64 version fails with an `No module named 'torchaudio.backend.common'` error (see screenshot 2). This should have been solved by the torchaudio team, but it seems like it was only fixed for arm64. Sorry for deleting 0.4.1 before I got positive feedback from anyone.

I will reupload 0.4.1 tomorrow (hopefully in the late morning) and next time keep some more version online. Sorry for the inconvenience.

![Bild 24 10 23 um 21 56](https://github.com/kaixxx/noScribe/assets/19264855/021793cf-482f-4091-8662-bf1c1dd265e0)

<img width="1212" alt="Bildschirmfoto 2023-10-24 um 22 13 49" src="https://github.com/kaixxx/noScribe/assets/19264855/3c66567b-a6eb-4df5-97fb-3da159ba6b05">

**Comment by BabyFnord** [comment]
Though the version I use now is 0.4.3, here's a suggestion for noScribe Edit to offer a different keyboard shortcut for playback. `⌘ + SPACE` defaults to the system-wide Spotlight search bar. `CTRL + SPACE` or `F5`/`F6` could be viable alternatives … thanks for considering.

**Reply by gernophil** [reply]
see comment https://github.com/kaixxx/noScribe/discussions/32#discussioncomment-7635184

**Comment by kaixxx** [comment]
> `CTRL + SPACE` or `F5`/`F6` could be viable alternatives

`CTRL + SPACE` is actually the shortcut I use on Windows and also the one I'm talking about in the documentation. So it would be good to use this on macOS also. Thank you for the suggestion.

@gernophil: In line 168 of noScribeEdit.py we set `self.play_along_action.setShortcut(QtGui.QKeySequence('Ctrl+Space'))`. Could you investigate why this translates to `⌘ + SPACE` on mac?

**Reply by gernophil** [reply]
I assume (= I am pretty sure) the reason for this is this:

> Note that on macOS, the Ctrl value corresponds to the Command keys on the Macintosh keyboard, and the Meta value corresponds to the Control keys.

Found [here](https://doc.qt.io/qt-5/qkeysequence.html#standard-shortcuts)

I also quite frequently use the `spotlight search` and it's a quite important shortcut. So I guess, one thing we could do is use `'Meta+Space'` for macOS or generally map '`Meta`' to '`Ctrl`'. if that's possible. However, I am not sure, if this would be a good idea.

**Reply by kaixxx** [reply]
> Meta value corresponds to the Control keys

Good find. So I would suggest changing the shortcut on macOS to 'Meta+Space' (not very urgent, can wait until the next release).

**Comment by maso73** [comment]
Hi there,
I was trying to install noscribe on macBook air (Intel, 9 years old) using the most current versions. I was following the instructions for Intel-based Macs but after confirming to run it in the gatekeeper I was not able to start noscribe again. noscribeEdit was working at least for version 0.42 . I have only 4GB RAM, could this be a reason for not starting? What else can I do? Any help would be appreciated.

**Reply by BabyFnord** [reply]
Please file a bug report under `Issues`, including your hardware and OS details and the `Console` output exclusive to `noScribe` and/or `noScribeEdit`, as it fails to start.
